src/System.Text.Json/tests/Serialization/NullableTests.cs

```diff
@@ -0,0 +1,253 @@
+                dictOfDictWithValue: new Dictionary<string, Dictionary<string, float?>> { { "key", dictWithFloatValue } },
+                dictOfDictWithNull: new Dictionary<string, Dictionary<string, float?>> { { "key", dictWithFloatNull } },
+                42.0f);
+
+            DateTime now = DateTime.Now;
+            Dictionary<string, DateTime?> dictWithDateTimeValue = new Dictionary<string, DateTime?> { { "key", now } };
+            Dictionary<string, DateTime?> dictWithDateTimeNull = new Dictionary<string, DateTime?> { { "key", null } };
+            TestDictionaryWithNullableValue<Dictionary<string, DateTime?>, Dictionary<string, Dictionary<string, DateTime?>>, DateTime?>(
+                dictWithDateTimeValue,
+                dictWithDateTimeNull,
+                dictOfDictWithValue: new Dictionary<string, Dictionary<string, DateTime?>> { { "key", dictWithDateTimeValue } },
+                dictOfDictWithNull: new Dictionary<string, Dictionary<string, DateTime?>> { { "key", dictWithDateTimeNull } },
+                now);
+
+            IDictionary<string, DateTime?> idictWithDateTimeValue = new Dictionary<string, DateTime?> { { "key", now } };
+            IDictionary<string, DateTime?> idictWithDateTimeNull = new Dictionary<string, DateTime?> { { "key", null } };
+            TestDictionaryWithNullableValue<IDictionary<string, DateTime?>, IDictionary<string, IDictionary<string, DateTime?>>, DateTime?>(
+                idictWithDateTimeValue,
+                idictWithDateTimeNull,
+                dictOfDictWithValue: new Dictionary<string, IDictionary<string, DateTime?>> { { "key", idictWithDateTimeValue } },
+                dictOfDictWithNull: new Dictionary<string, IDictionary<string, DateTime?>> { { "key", idictWithDateTimeNull } },
+                now);
+
+            ImmutableDictionary<string, DateTime?> immutableDictWithDateTimeValue = ImmutableDictionary.CreateRange(new Dictionary<string, DateTime?> { { "key", now } });
+            ImmutableDictionary<string, DateTime?> immutableDictWithDateTimeNull = ImmutableDictionary.CreateRange(new Dictionary<string, DateTime?> { { "key", null } });
+            TestDictionaryWithNullableValue<ImmutableDictionary<string, DateTime?>, ImmutableDictionary<string, ImmutableDictionary<string, DateTime?>>, DateTime?>(
+                immutableDictWithDateTimeValue,
+                immutableDictWithDateTimeNull,
+                dictOfDictWithValue: ImmutableDictionary.CreateRange(new Dictionary<string, ImmutableDictionary<string, DateTime?>> { { "key", immutableDictWithDateTimeValue } }),
+                dictOfDictWithNull: ImmutableDictionary.CreateRange(new Dictionary<string, ImmutableDictionary<string, DateTime?>> { { "key", immutableDictWithDateTimeNull } }),
+                now);
+        }
+
+        public class MyOverflowWrapper
+        {
+            [JsonExtensionData]
+            public Dictionary<string, JsonElement?> MyOverflow { get; set; }
```

Comment:
Probably unrelated, but if you had another property with `JsonExtensionData` attribute (which doesn't have nullable) would we detect that and fail?

Say:
```C#
[JsonExtensionData]
public Dictionary<string, JsonElement?> MyOverflow { get; set; }
[JsonExtensionData]
public Dictionary<string, JsonElement> MyOverflow { get; set; }
```

Reply:
Yes we would: https://github.com/dotnet/corefx/blob/5e102fb5d30af4bc940aa519195c0e3dbd933648/src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs#L248-L251.

I added tests for this scenario and the reverse order (valid overflow first).